cms/core/models/base.py

```diff
@@ -79,9 +83,13 @@ def has_ons_embed(self) -> bool:
         """Checks if there are any ONS embed blocks.
         Override in your specific Page class if the StreamField structure is different.
         """
-        if streamfield := getattr(self, self.content_field_name):
+        if streamvalue := getattr(self, self.content_field_name):
+            # run the check on the StreamBlock itself, if it supports it
+            if hasattr(streamvalue.stream_block, "has_ons_embed"):
+                return bool(streamvalue.stream_block.has_ons_embed(streamvalue))
+
             try:
-                return streamfield.first_block_by_name(block_name="ons_embed") is not None
+                return streamvalue.first_block_by_name(block_name="ons_embed") is not None
```

Comment:
Same here. I think we should embrace the new API and return `False` if not implemented.

Reply:
68232d5